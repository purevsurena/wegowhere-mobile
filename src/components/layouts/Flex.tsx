import React from 'react';
import { View } from 'react-native';
import { FlexProps } from '@/types/layouts';

function Flex(props: FlexProps): JSX.Element {
  const { flex, fd, as, ai, jc, fw, style, children } = props;

  return (
    <View
      style={{
        flex: flex,
        alignSelf: as,
        alignItems: ai,
        justifyContent: jc,
        flexDirection: fd,
        flexWrap: fw,
        ...style,
      }}>
      {children}
    </View>
  );
}

export default Flex;
