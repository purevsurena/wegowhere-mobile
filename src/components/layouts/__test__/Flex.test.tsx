import React from 'react';
import renderer from 'react-test-renderer';
import Flex from '@/components/layouts/Flex';
import { Text } from 'react-native-svg';

describe('<Flex />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Flex flex={1} fd="row" ai="center" jc="space-between">
          <Text>Child 1</Text>
        </Flex>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
