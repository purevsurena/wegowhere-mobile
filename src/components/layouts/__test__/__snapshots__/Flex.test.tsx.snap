// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Flex /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "alignSelf": undefined,
      "flex": 1,
      "flexDirection": "row",
      "flexWrap": undefined,
      "justifyContent": "space-between",
    }
  }
>
  <RNSVGText
    content={null}
    dx={[]}
    dy={[]}
    fill={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
    font={{}}
    rotate={[]}
    x={[]}
    y={[]}
  >
    <RNSVGTSpan
      content="Child 1"
      dx={[]}
      dy={[]}
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      font={{}}
      rotate={[]}
      x={[]}
      y={[]}
    />
  </RNSVGText>
</View>
`;
